src/main/java/ai/autonomic/executor/humdinger/HumdingerController.java

```
@@ -0,0 +1,103 @@
+/*
+ * Autonomic Proprietary 1.0
+ *
+ * Copyright (c) 2016, Autonomic Incorporated - All rights reserved
+ *
+ * Proprietary and confidential.
+ *
+ * NOTICE:  All information contained herein is, and remains the property of
+ * Autonomic Incorporated and its suppliers, if any.  The intellectual and
+ * technical concepts contained herein are proprietary to Autonomic Incorporated
+ * and its suppliers and may be covered by U.S. and Foreign Patents, patents in
+ * process, and are protected by trade secret or copyright law. Dissemination of
+ * this information or reproduction of this material is strictly forbidden unless
+ * prior written permission is obtained from Autonomic Incorporated.
+ *
+ * Unauthorized copy of this file, via any medium is strictly prohibited.
+ *
+ */
+package ai.autonomic.executor.humdinger;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+
+@RestController
+public class HumdingerController {
+  private static final Logger logger = LoggerFactory.getLogger(HumdingerController.class);
+  private final AtomicLong counter = new AtomicLong();
+  @Autowired
+  private MDIData mdi_service;
+
+
+  @RequestMapping("/mdi-test")
+  public Response greeting_two(@RequestParam(value="name", defaultValue="World") String name) {
+    Map[] data = mdi_service.get_url("api/v2/modules/versions");
```

What kind of data is in your `Map`? Use generics. Is it a `Map<String,String>`?
Also, as mentioned below, `get_url` (change this name), should raise an Exception if something goes wrong. Otherwise, you'll get a `NullPointerException` here and you won't know why at first. (And there's no test for it and no logging.) And it might not be your fault.